### PR TITLE
chore: update to cosmos-sdk v0.45.9

### PR DESCRIPTION
**Pull request by aaronc**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by anilCSE — APPROVED** [review]
We should merge this into master too right?

**Comment by aaronc** [comment]
> We should merge this into master too right?

Sure we can

**Review by clevinson — CHANGES_REQUESTED** [review]
We need to upgrade our fork of v0.45 I think, instead of just bumping here.

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/CosmWasm/wasmd v0.22.0
-	github.com/cosmos/cosmos-sdk v0.45.4
+	github.com/cosmos/cosmos-sdk v0.45.9
```
Comment (aleem1314) [inline go.mod:7]: Is it okay to not update our SDK fork? We are using [SDK](https://github.com/regen-network/regen-ledger/pull/1549/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L151) forked version.

**Comment by clevinson** [comment]
I'm updating our fork, will update here when that's done so we can update this PR accordingly.

**Comment by aaronc** [comment]
Updated to the tag on our fork. Can you re-review @clevinson ?

File: go.mod [hunk 2]
```diff
@@ -158,3 +161,5 @@ replace github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.7.0
 
 // Mock wasmd module removed in experimental build
 replace github.com/CosmWasm/wasmd => ./mocks/wasmd
+
+replace github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
```
Comment (clevinson) [inline go.mod:165]: @aaronc Does this need to also come from our SDK fork? or is that not necessary.

Comment (clevinson) [reply go.mod:165]: OK just double checked the `ics23/go/v0.8.0` tag on SDK upstream. Looks correct to me.

**Review by clevinson — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -148,7 +151,7 @@ replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-replace github.com/cosmos/cosmos-sdk => github.com/regen-network/cosmos-sdk v0.45.0-regen-1
+replace github.com/cosmos/cosmos-sdk => github.com/regen-network/cosmos-sdk v0.45.9-regen-1
 
 replace github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.14
```
Comment (aaronc) [inline go.mod:156]: Build failing because of this line. Why do we need it?

**Pull request merged**